tests/optvis/test_objectives.py

```diff
@@ -41,6 +41,37 @@ def test_neuron(inceptionv1):
     objective = objectives.neuron("mixed4a_pre_relu", 42)
     assert_gradient_ascent(objective, inceptionv1)
 
+def test_composition():
+    @wrap_objective
+    def f(a):
+      return lambda T: a
+
+    a   = f(1)
+    b   = f(2)
+    c   = f(3)
+    ab  = a - 2*b
+    cab = c*(ab - 1)
+
+    assert str(cab)  == "F(3)·((F(1) + F(2)·2·-1) + -1)"
+    assert cab(None) == 3*(1 - 2*2 - 1)
+    assert a.value   == 1
+    assert b.value   == 2
+    assert c.value   == 3
+    assert ab.value  == (a.value - 2*b.value)
+    assert cab.value == c.value*(ab.value - 1)
+
+
+@pytest.mark.parametrize("cossim_pow", [0, 1, 2])
+def test_cossim():
```

Comment:
I believe you need to add `cossim_pow` to the arguments of this test. Did this actually run without it? O_o